hikari/impl/buckets.py

```diff
@@ -590,18 +626,34 @@ def update_rate_limits(
                 limit_header,
                 remaining_header,
             )
-        else:
-            bucket = RESTBucket(real_bucket_hash, compiled_route)
+        except KeyError:
+            try:
```

Comment:
just a general question but wouldn't it be cleaner and more explicit to do 
```py
if bucket := ...
```
here instead of the try and accept?

Reply:
try and catching the errors seems to be a bit faster then using a walrus and if/else statements

Reply:
and is it still faster when an error is raised

Reply:
Yep, both cases

Reply:
even with all that over head from error handling?

Reply:
Overhead is really performance hurtful when using `contextlib.suppress` (which I found some uses off and need to remove). A bare try/except doesn't seem to have any performance implication and seems to be even faster than using an if/else clause

Reply:
So you're fine with putting micro-optimisations over readability?

Reply:
This doesn't make the code super unreadable. But I can change it back to if/else statements if you think its better

Reply:
> This doesn't make the code super unreadable. 
it's more ambiguous as to what case catching KeyError is handling compared to if := since there's multiple function calls in the try section which could all be raising it